src/shazamkit.cs

```diff
@@ -0,0 +1,267 @@
+using System;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+using AVFoundation;
+
+namespace ShazamKit {
+
+	[Native]
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[ErrorDomain ("SHErrorDomain")]
+	public enum SHErrorCode : long
+	{
+		InvalidAudioFormat = 100,
+		AudioDiscontinuity = 101,
+		SignatureInvalid = 200,
+		SignatureDurationInvalid = 201,
+		MatchAttemptFailed = 202,
+		CustomCatalogInvalid = 300,
+		CustomCatalogInvalidURL = 301,
+		MediaLibrarySyncFailed = 400,
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[Static]
+	enum SHMediaItemProperty
+	{
+		[Field ("SHMediaItemShazamID")]
+		ShazamId,
+		[Field ("SHMediaItemTitle")]
+		Title,
+		[Field ("SHMediaItemSubtitle")]
+		Subtitle,
+		[Field ("SHMediaItemArtist")]
+		Artist,
+		[Field ("SHMediaItemWebURL")]
+		WebUrl,
+		[Field ("SHMediaItemAppleMusicID")]
+		AppleMusicId,
+		[Field ("SHMediaItemAppleMusicURL")]
+		AppleMusicUrl,
+		[Field ("SHMediaItemArtworkURL")]
+		ArtworkUrl,
+		[Field ("SHMediaItemVideoURL")]
+		VideoUrl,
+		[Field ("SHMediaItemExplicitContent")]
+		ExplicitContent,
+		[Field ("SHMediaItemGenres")]
+		Genres,
+		[Field ("SHMediaItemISRC")]
+		Isrc,
+		[Field ("SHMediaItemMatchOffset")]
+		MatchOffset,
+		[Field ("SHMediaItemFrequencySkew")]
+		FrequencySkew,
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHCatalog
+	{
+		[Export ("minimumQuerySignatureDuration")]
+		double MinimumQuerySignatureDuration { get; }
+
+		[Export ("maximumQuerySignatureDuration")]
+		double MaximumQuerySignatureDuration { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHCatalog))]
+	interface SHCustomCatalog
+	{
+		[Export ("addReferenceSignature:representingMediaItems:error:")]
+		bool Add (SHSignature signature, SHMediaItem[] mediaItems, [NullAllowed] out NSError error);
+
+		[Export ("addCustomCatalogFromURL:error:")]
+		bool Add (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("writeToURL:error:")]
+		bool Write (NSUrl url, [NullAllowed] out NSError error);
+
+		[Static]
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
```

Comment:
^ that's a (incomplete) duplicate of the attributes on the type

Reply:
Good catch! Removing now!